File: samples/inventory-service/inventory-py/inventory/reservation.py
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum


class ReservationStatus(Enum):
    """Status of a stock reservation."""

    PENDING = "pending"
    CONFIRMED = "confirmed"
    EXPIRED = "expired"


@dataclass
class ReservationRequest:
    """A request to hold inventory for an order."""

    sku: str
    quantity: int
    order_id: str


@dataclass
class ReservationResult:
    """Outcome of a reservation attempt."""

    reservation_id: str
    status: ReservationStatus
    expires_at: str

File: samples/inventory-service-with-predefined-models/inventory-go/types/reservation.go
package types

type ReservationStatus string

const (
	ReservationPending   ReservationStatus = "pending"
	ReservationConfirmed ReservationStatus = "confirmed"
	ReservationExpired   ReservationStatus = "expired"
)

// ReservationRequest asks to hold inventory for an order.
type ReservationRequest struct {
	SKU      string `json:"sku"`
	Quantity int    `json:"quantity"`
	OrderID  string `json:"orderId"`
}

// ReservationResult returns a claim for an order reservation.
type ReservationResult struct {
	ReservationID string            `json:"reservationId"`
	Status        ReservationStatus `json:"status"`
	ExpiresAt     string            `json:"expiresAt"`
}
